test/ttexalens/unit_tests/test_risc_debug.py

```
@@ -270,6 +270,76 @@ def test_minimal_run_generated_code(self):
+        l1_addr = 0x1000
+        samples = 1
+        sampling_interval = 2
+        group_values = signal_store.read_signal_group(
+            group_name, l1_addr, samples=samples, sampling_interval=sampling_interval
+        )
+        assert pc_signal_name in group_values.keys()
+
+        assert group_values[pc_signal_name][0] == pc_value_32
+
+        # check all groups for chosen core
+        for group in signal_store.debug_bus_signals.group_names:
+            # skip groups that do not belong to this core
+            if not group.startswith(self.core_sim.risc_name.lower() + "_"):
+                continue
+
+            l1_addr = 0x1000
+            samples = 3
+            group_values = signal_store.read_signal_group(group, l1_addr, samples=samples)
+
+            for signal_name, values in group_values.items():
+                parts = signal_store.debug_bus_signals.get_signal_part_names(signal_name)
+
+                if signal_store.debug_bus_signals.is_combined_signal(signal_name):
+                    # combined signal
+                    combined_value = 0
+                    for part in parts:
+                        part_value = signal_store.read_signal(part)
+                        part_description = signal_store.get_signal_description(part)
+                        shift = (part_description.mask & -part_description.mask).bit_length() - 1
+                        combined_value |= part_value << shift
+
+                    combined_value = signal_store._normalize_value(
+                        combined_value, signal_store.get_signal_description(parts[0]).mask
+                    )
+                    assert values[0] == combined_value, f"Combined signal {signal_name} value mismatch."
+                else:
+                    # single part signal
+                    signal_value_32 = signal_store.read_signal(signal_name)
+                    assert values[0] == signal_value_32, f"Signal {signal_name} value mismatch."
+        return
+
```

We should also add invalid tests to check if we raise exceptions for invalid arguments.